rocAL/rocAL/source/bounding_box_graph.cpp

```diff
@@ -147,3 +147,159 @@ void BoundingBoxGraph::update_random_bbox_meta_data(MetaDataBatch *input_meta_da
+            _anchor.l = anchors[m];
+            _anchor.t = anchors[m + 1];
+            _anchor.r = anchors[m + 2];
+            _anchor.b = anchors[m + 3];
+            const auto best_idx = find_best_box_for_anchor(anchor_idx, ious, bb_count, anchors_size);
+            // Filter matches by criteria
+            if (ious[(best_idx * anchors_size) + anchor_idx] > criteria) //Its a match
+            {
+                //YTD: Need to add a new structure for xc,yc,w,h similar to l,t,r,b as a part of metadata
+                    box_bestidx.l = 0.5 * (bb_coords.at(best_idx).l + bb_coords.at(best_idx).r); //xc
+                    box_bestidx.t = 0.5 * (bb_coords.at(best_idx).t + bb_coords.at(best_idx).b); //yc
+                    box_bestidx.r = (-bb_coords.at(best_idx).l + bb_coords.at(best_idx).r);      //w
+                    box_bestidx.b = (-bb_coords.at(best_idx).t + bb_coords.at(best_idx).b);      //h
+
+                if (offset)
+                {
+                    box_bestidx.l *= scale; //xc
+                    box_bestidx.t *= scale; //yc
+                    box_bestidx.r *= scale; //w
+                    box_bestidx.b *= scale; //h
+                    //YTD: Need to add a new structure for xc,yc,w,h similar to l,t,r,b as a part of metadata
+                    _anchor_xcyxwh.l = 0.5 * (_anchor.l + _anchor.r) * scale; //xc
+                    _anchor_xcyxwh.t = 0.5 * (_anchor.t + _anchor.b) * scale; //yc
+                    _anchor_xcyxwh.r = (-_anchor.l + _anchor.r) * scale;      //w
+                    _anchor_xcyxwh.b = (-_anchor.t + _anchor.b) * scale;      //h
+
+                    box_bestidx.l = ((box_bestidx.l - _anchor_xcyxwh.l) / _anchor_xcyxwh.r - means[0]) / stds[0];
+                    box_bestidx.t = ((box_bestidx.t - _anchor_xcyxwh.t) / _anchor_xcyxwh.b - means[1]) / stds[1];
+                    box_bestidx.r = (std::log(box_bestidx.r / _anchor_xcyxwh.r) - means[2]) / stds[2];
+                    box_bestidx.b = (std::log(box_bestidx.b / _anchor_xcyxwh.b) - means[3]) / stds[3];
+                    encoded_bb[anchor_idx] = box_bestidx;
+                    encoded_labels[anchor_idx] = bb_labels.at(best_idx);
+                }
+                else
+                {
+                    encoded_bb[anchor_idx] = box_bestidx;
+                    encoded_labels[anchor_idx] = bb_labels.at(best_idx);
+                }
+            }
+            else // Not a match
+            {
+                if (offset)
+                {
+                    _anchor_xcyxwh.l =_anchor_xcyxwh.t = _anchor_xcyxwh.r = _anchor_xcyxwh.b = 0;
+                    encoded_bb[anchor_idx] = _anchor_xcyxwh;
+                    encoded_labels[anchor_idx] = 0;
+                }
+                else
+                {
+                    //YTD: Need to add a new structure for xc,yc,w,h similar to l,t,r,b as a part of metadata
+                    _anchor_xcyxwh.l = 0.5 * (_anchor.l + _anchor.r); //xc
```

Comment:
store both formats to avoid conversions in critical path

Reply:
Done, storing in xcycwh format